common/strings.c

```diff
@@ -169,6 +169,85 @@ TDNFSplitStringToArray(
     goto cleanup;
 }
 
+uint32_t
+TDNFMergeStringArrays(
+    char ***pppszArray0,
+    char **ppszArray1
+)
+{
+    uint32_t dwError = 0;
+    int i, n, n0, n1;
+
+    if (!pppszArray0 || !ppszArray1) {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    dwError = TDNFStringArrayCount(*pppszArray0, &n0);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    dwError = TDNFStringArrayCount(ppszArray1, &n1);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    n = n0 + n1;
+
+    dwError = TDNFReAllocateMemory((n + 1) * sizeof(char *), (void **)pppszArray0);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    for (i = 0; i < n1; i++) {
+        (*pppszArray0)[n0 + i] = ppszArray1[i];
+    }
+    (*pppszArray0)[n] = NULL;
+
+    /* intentionally free only the pointer list,
+       and not the strings, because those have been moved to
+       pppszArray0 */
+    TDNF_SAFE_FREE_MEMORY(ppszArray1);
```

Comment:
Once this gets freed, what happens to `pppszArray0` ? at line 198 it is just pointer assignment right?

Reply:
Yes. The ownership for the strings gets transfered from `ppszArray1` to `pppszArray0`. The latter is owned by the caller.